src/util/safe_pointer.h

```diff
@@ -0,0 +1,56 @@
+/*******************************************************************\
+
+Module: Simple checked pointers
+
+Author: Chris Smowton, chris@smowton.net
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_SAFE_POINTER_H
+#define CPROVER_UTIL_SAFE_POINTER_H
+
+template<class T> class safe_pointer
+{
+ public:
+  operator bool()
+  {
+    return !!ptr;
+  }
+
+  T *get()
+  {
+    if(!(*this))
```

Comment:
I'd prefer an assertion here, because it's clearly a programming error and not something that someone should be able to catch.